File: C++ (g++) 6.2/Catch/hiker.tests.cpp
// For more on how to use the excellent Catch test framework see
// https://github.com/philsquared/Catch

#include "hiker.hpp"
#include "catch.hpp"

TEST_CASE( "Life the universe and everything", "[hhgttg]" )
{
    REQUIRE( answer() == 42 );
}

File: swift-swordfish/hikertests.swift
// For more information on using the excellent Swordfish test framework see
// https://github.com/philsquared/Swordfish

class HikerTests: SwordfishTests {
  var hiker : Hiker!

  override func setUp() {
    super.setUp()
    hiker = Hiker()
  }

  func testLife_the_universe_and_everything() {
    require() | hiker.answer() == 42
  }
}

extension HikerTests {
    static var allTests : [(String, (HikerTests) -> () throws -> Void)] {
        return [
            ("testLife_the_universe_and_everything", testLife_the_universe_and_everything),
        ]
    }
}
